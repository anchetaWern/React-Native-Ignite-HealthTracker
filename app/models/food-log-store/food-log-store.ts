import { Instance, SnapshotIn, SnapshotOut, types } from "mobx-state-tree"

import { FoodLogModel, FoodLogSnapshotIn } from "../food-log/food-log"

/**
 * Model description here for TypeScript hints.
 */
export const FoodLogStoreModel = types
  .model("FoodLogStore")
  .props({
    food_logs: types.optional(types.array(FoodLogModel), []),
  })
  .views((self) => ({
    get allLogs() {
      return self.food_logs
    },
  })) // eslint-disable-line @typescript-eslint/no-unused-vars
  .actions((self) => ({
    saveLog: (foodLogSnapshot: FoodLogSnapshotIn) => {
      self.food_logs.push(foodLogSnapshot)
    },
  })) // eslint-disable-line @typescript-eslint/no-unused-vars

export interface FoodLogStore extends Instance<typeof FoodLogStoreModel> {}
export interface FoodLogStoreSnapshotOut extends SnapshotOut<typeof FoodLogStoreModel> {}
export interface FoodLogStoreSnapshotIn extends SnapshotIn<typeof FoodLogStoreModel> {}
export const createFoodLogStoreDefaultModel = () => types.optional(FoodLogStoreModel, {})

/*
 food_logs: types.optional(types.array(FoodLogModel), []),
*/
