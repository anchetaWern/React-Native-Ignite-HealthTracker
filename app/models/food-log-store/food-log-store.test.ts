import { FoodLogStoreModel } from "./food-log-store"

test("can be created", () => {
  const instance = FoodLogStoreModel.create({
    food_logs: [
      {
        id: "someid123",
        food_id: "somefoodid123",
        rating: 5,
        date: new Date(),
      },
    ],
  })

  expect(instance).toBeTruthy()
})
