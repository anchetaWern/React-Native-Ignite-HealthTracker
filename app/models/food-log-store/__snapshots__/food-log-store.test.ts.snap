// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can be created 1`] = `
Object {
  "food_logs": Array [
    Object {
      "date": 1664082280927,
      "food_id": "somefoodid123",
      "id": "someid123",
      "rating": 5,
    },
  ],
}
`;
